src/main/java/net/neoforged/neoforge/client/event/GatherSkippedAttributeTooltipsEvent.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.EquipmentSlotGroup;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.Event;
+import net.neoforged.neoforge.common.util.AttributeTooltipContext;
+
+/**
+ * This event is used to collect the IDs of attribute modifiers that will not be displayed in item tooltips.
+ * <p>
+ * It allows hiding some (or all) of the modifiers, potentially for displaying them in an alternative way (or for hiding information from the player).
+ * <p>
+ * This event may be fired on both the logical client and logical server.
+ */
+public class GatherSkippedAttributeTooltipsEvent extends Event {
+    protected final ItemStack stack;
+    protected final Set<ResourceLocation> skippedIds;
+    protected final Set<EquipmentSlotGroup> skippedGroups;
+    protected final AttributeTooltipContext ctx;
+    protected boolean skipAll = false;
+
+    public GatherSkippedAttributeTooltipsEvent(ItemStack stack, AttributeTooltipContext ctx) {
+        this.stack = stack;
+        this.skippedIds = new HashSet<>();
+        this.skippedGroups = EnumSet.noneOf(EquipmentSlotGroup.class);
```

Comment:
As these are not used outside of this class, could we make them private, and then lazily initialize them? That way we don't have unnecessary object churn and initialization costs? Especially given in the majority of cases these will be unused.

Reply:
Yeah we can.  Not sure how much it matters in the grand scheme of things, but it might help.